include/mbedtls/rsa.h

```diff
@@ -38,6 +38,60 @@
 #include "mbedtls/threading.h"
 #endif
 
+#if defined(PSA_CRYPTO_DRIVER_TEST)
+#include <psa/crypto_driver_common.h>
+#include <psa/crypto.h>
+
+typedef struct {
+    /* If non-null, on success, copy this to the output. */
+    void *forced_output;
+    size_t forced_output_length;
```

Comment:
Shouldn't all of this code be inside `extern "C" {`?

Reply:
Function declarations must be within `extern "C" { … }` in public headers for the sake of C++ compilers. This only applies to public headers: we don't aim for the library code, the test code or the sample programs to be valid C++ programs.